public/app/features/dashboard/components/Inspector/InspectContent.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import { getLocationSrv } from '@grafana/runtime';
+import { getPanelInspectorStyles } from './styles';
+import { CustomScrollbar, Drawer, TabContent } from '@grafana/ui';
+import { InspectSubtitle } from './InspectSubtitle';
+import { InspectDataTab } from './InspectDataTab';
+import { InspectMetadataTab } from './InspectMetadataTab';
+import { InspectJSONTab } from './InspectJSONTab';
+import { InspectErrorTab } from './InspectErrorTab';
+import { InspectStatsTab } from './InspectStatsTab';
+import { QueryInspector } from './QueryInspector';
+import { InspectTab } from './types';
+import { DashboardModel, PanelModel } from '../../state';
+import { DataSourceApi, PanelData, PanelPlugin } from '@grafana/data';
+import { GetDataOptions } from '../../state/PanelQueryRunner';
+
+interface Props {
+  dashboard: DashboardModel;
+  panel: PanelModel;
+  plugin?: PanelPlugin | null;
+  defaultTab: InspectTab;
+  tabs: Array<{ label: string; value: InspectTab }>;
+  // The last raw response
+  data?: PanelData;
+  isDataLoading: boolean;
+  dataOptions: GetDataOptions;
+  onDataOptionsChange: (options: GetDataOptions) => void;
+  // If the datasource supports custom metadata
+  metadataDatasource?: DataSourceApi;
+}
+export const InspectContent: React.FC<Props> = ({
+  panel,
+  plugin,
+  dashboard,
+  tabs,
+  data,
+  isDataLoading,
+  dataOptions,
+  onDataOptionsChange,
+  metadataDatasource,
+  defaultTab,
+}) => {
+  const [currentTab, setCurrentTab] = useState(defaultTab ?? InspectTab.Data);
+
+  if (!plugin) {
+    return null;
+  }
+
+  const onClose = () => {
+    getLocationSrv().update({
+      query: { inspect: null, inspectTab: null },
+      partial: true,
+    });
+  };
```

Comment:
Interesting, any reason we're not using the `updateLocation` action for this?

Reply:
Totally no reason. Just muscle memory. Good point @hugohaggmark, I'm going to update this :)